tests/tracking/test_tracking.py

```diff
@@ -134,6 +134,67 @@ def _assert_exps(ids_to_lifecycle_stage, view_type_arg):
     _assert_exps({"1": LifecycleStage.DELETED}, ViewType.DELETED_ONLY)
 
 
+def test_list_experiments_paginated():
+    experiments = []
+    for i in range(10):
+        experiments.append(mlflow.create_experiment("paginated_exp_" + str(i)))
+    max_results = 5
+    returned_experiments = []
+    client = tracking.MlflowClient()
+    result = client.list_experiments(max_results=max_results, page_token=None)
+    assert len(result) == max_results
+    returned_experiments.extend(result)
+    while result.token:
+        result = client.list_experiments(max_results=max_results, page_token=result.token)
+        assert len(result) <= max_results
+        returned_experiments.extend(result)
+    assert result.token is None
+    returned_exp_id_set = set([exp.experiment_id for exp in returned_experiments])
+    assert set(experiments) - returned_exp_id_set == set()
+
+
+def test_list_experiments_paginated_returns_in_correct_order():
```

Comment:
@smurching @wamartin-aml 

I found that this test fails with `FileStore`.

Code to reproduce:

```suggestion
@pytest.mark.notrackingurimock
def test_list_experiments_paginated_returns_in_correct_order(tmpdir):
    mlflow.set_tracking_uri(tmpdir.strpath)
```

Is this what we intend?